src/Form/SubscribeForm.php

```diff
@@ -188,14 +227,34 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
       ];
     }
 
-    // Add subscriber(s).
-    $results = [];
+    $new_subscription_lists = $subscribe_lists;
+
+    // Determine which lists the user has already subscribed to.
+    $subscriber_id = $this->apsis->getSubscriberIdByEmail($email);
+    if (!empty($subscriber_id)) {
+      $subscriber_lists = $this->apsis->getSubscriberMailingLists($subscriber_id);
```

Comment:
Note that this is missing error setting of short timeouts to avoid causing timeouts before the subscription is put on the queue.

Reply:
With 574912c232b3a72418ba01a5f62561719978a40e this is now complete.